src/ucp/amo/amo.inl

```diff
@@ -0,0 +1,207 @@
+/**
```

Comment:
I think most of this file can be moved to nb_amo.c. I see that only UCP_AMO_WITHOUT_RESULT and ucp_amo_check_send_status() are being used by both basic and nb amos